lib/src/test/java/com/dynatrace/metric/util/DynatraceMetadataEnricherTest.java

```diff
@@ -19,50 +19,64 @@
 import java.io.FileReader;
 import java.io.IOException;
 import java.io.StringReader;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.List;
+import java.util.*;
 import org.junit.jupiter.api.Test;
 import org.mockito.MockedStatic;
 import org.mockito.Mockito;
 
-public class DynatraceMetadataEnricherTest {
+class DynatraceMetadataEnricherTest {
 
   @Test
-  public void validMetrics() {
+  void validProperties() {
+    Properties properties = new Properties();
+    properties.setProperty("prop.a", "value.a");
+    properties.setProperty("prop.b", "value.b");
+
     ArrayList<Dimension> entries =
-        new ArrayList<>(
-            DynatraceMetadataEnricher.parseDynatraceMetadata(
-                Arrays.asList("prop.a=value.a", "prop.b=value.b")));
-
-    assertEquals("prop.a", entries.get(0).getKey());
-    assertEquals("value.a", entries.get(0).getValue());
-    assertEquals("prop.b", entries.get(1).getKey());
-    assertEquals("value.b", entries.get(1).getValue());
+        new ArrayList<>(DynatraceMetadataEnricher.createDimensionList(properties));
+
+    // Has one entry with key "prop.a"
+    assertEquals(1, entries.stream().filter(entry -> entry.getKey().equals("prop.a")).count());
```

Comment:
Maybe we could just access the entries by their index? instead of repeating the `stream()` portion? 🤔

Reply:
The order of the entries is now dependent on the hash of the key. So using a different key in the setup would cause the test to potentially fail when accessing it. 

Do you think adding a comment for that and changing it to use indexes would be better? :thinking:

Reply:
Ah I see. Then if it could cause flakiness let's keep this way.